Source/WebCore/platform/graphics/OwnedDecodedImage.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "OwnedDecodedImage.h"
+
+namespace WebCore {
+
+std::unique_ptr<OwnedDecodedImage> OwnedDecodedImage::create(PlatformImagePtr&& image)
+{
+    if (!image)
+        return nullptr;
+
+    return makeUnique<OwnedDecodedImage>(WTFMove(image));
+}
+
+OwnedDecodedImage::OwnedDecodedImage(PlatformImagePtr&& image)
+    : DecodedImage(DecodedImageType::Owned)
+    , m_platformImage(WTFMove(image))
+{
+    ASSERT(m_platformImage);
+}
+
+OwnedDecodedImage::~OwnedDecodedImage() = default;
+
+PlatformImagePtr OwnedDecodedImage::platformImage() const
+{
+    return m_platformImage;
+}
+
+std::unique_ptr<CachedDecodedImage> OwnedDecodedImage::createCachedDecodedImage() const
+{
+    return nullptr;
+}
```

Comment:
I think DecodedImage::createCachedDecodedImage() should not be pure virtual. It should just return a nullptr and we can get rid of this one.